src/services/metadata-service.ts

```diff
@@ -54,10 +55,14 @@ export class MetadataService implements IMetadataService {
     private readonly ipfsService: IIpfsService
   ) {}
 
+  async fill(streamId: StreamID, genesisFields: GenesisFields): Promise<void> {
+    await this.storeMetadata(streamId, genesisFields)
+  }
+
   /**
    * Retrieve genesis header fields from IPFS, store to the database.
    */
-  async fill(streamId: StreamID, options: AbortOptions = {}): Promise<void> {
+  async fillFromIpfs(streamId: StreamID, options: AbortOptions = {}): Promise<void> {
     const isPresent = await this.metadataRepository.isPresent(streamId)
     if (isPresent) return // Do not perform same work of retrieving from IPFS twice
     const genesisFields = await this.retrieveFromGenesis(streamId, options)
```

Comment:
Oh, sorry for not figuring out it earlier. `options` could be `AbortOptions & {genesisFields?: GenesisFields}`. Then almost the same `fill` function would do the trick.

```
const genesisFields = options.genesisFields || await this.retrieveFromGenesis(streamId, options)
```

Or, `if`-statement could be better: it would contain two different log messages: one for passed options, another if it is retrieved from ipfs.

Reply:
Hmm, nope. Bad suggestion. Please, disregard.